website/content/posts/evaluating-custom-predicates.md

```diff
@@ -0,0 +1,196 @@
+---
+title: "Policies Polyglot: Evaluating Custom Predicates"
+date: 2024-03-20T15:02:00-04:00
+author: "Luiz Carvalho"
+---
+
+Attestations are a wonderful way to attach metadata to container images in a secure manner. One of
+the most popular format is [SLSA Provenance](https://slsa.dev/spec/v0.1/provenance#schema) which is
+used to provide information on how the image created. Our [Hitchhiker’s
+Guide](https://enterprisecontract.dev/docs/user-guide/main/hitchhikers-guide.html) demonstrates how
+to write policies to assert the contents of the SLSA Provenance. Here, we expand on that approach to
+assert the contents of *any* attestation format, even completely made up ones.
+
+<!--more-->
+
+Before getting started, let's make sure we have an image that is already signed and has a SLSA
+Provenance attestation. We will also need access to the signing key used. The [Hitchhiker’s
+Guide](https://enterprisecontract.dev/docs/user-guide/main/hitchhikers-guide.html) walks through the
+process. If you want to try out the commands in this blog post, start there.
+
+When we talk about different attestation formats, what we are really saying is different **predicate
+types** of an in-toto attestation. In addition to SLSA Provenance, there are a few [other common
+ones](https://github.com/in-toto/attestation/blob/main/spec/predicates/README.md). As suggested
+earlier, in this blog post we will use a made up predicate type to both demistify predicates and
+highlight Enterprise Contract's flexibility in validating attestations.
+
+Let's get this started!
+
+First, we create a `spam.json` file with arbitrary JSON data. This holds the contents of our custom
+predicate.
+
+```json
+{"spam": "more spam"}
```

Comment:
:shrug: 
```suggestion
{"bacon_style": "crispy", "bacon_count": 6}
```

Reply:
I know it's arbitrary, but maybe it's more relatable to have a key and value that means something concrete.

Reply:
I like it. Done